pkg/raft/term_cache_test.go

```diff
@@ -0,0 +1,505 @@
+// Copyright 2025 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package raft
+
+import (
+	"fmt"
+	"testing"
+
+	pb "github.com/cockroachdb/cockroach/pkg/raft/raftpb"
+	"github.com/stretchr/testify/require"
+)
+
+// Test constants, the test cases are hardcoded based on current test constants.
+var TestInitialTerm = uint64(5)
+var TestInitialIndex = uint64(10)
+
+func TestTermCacheInitialization(t *testing.T) {
+	tc := NewTermCache(8, entryID{term: 5, index: 10})
+
+	// Verify basic properties after initialization
+	require.Len(t, tc.cache, 1)
+	require.Equal(t, entryID{term: 5, index: 10}, tc.firstEntry())
+	require.Equal(t, uint64(10), tc.cache[0].index)
+	require.Equal(t, 8, cap(tc.cache))
+}
+
+func TestTermCacheTerm(t *testing.T) {
+	// Create a term cache with initial entry at term=5, index=10
+	initialID := entryID{term: 5, index: 10}
+	tc := NewTermCache(4, initialID)
+
+	// Create a log slice with entries that have different term patterns
+	// This creates entries at indices 11-14 with terms [5,6,6,7]
+	leadSlice := initialID.append(5, 6, 6, 7)
+	tc.truncateAndAppend(leadSlice.LogSlice)
+
+	// Test term lookup for various indices
+	testCases := []struct {
+		index    uint64
+		wantTerm uint64
+		wantOK   bool
+	}{
+		{9, 0, false}, // below cache
+		{10, 5, true}, // initial entry
+		{11, 5, true}, // same term continuation
+		{12, 6, true}, // term flip
+		{13, 6, true}, // same term
+		{14, 7, true}, // another term flip
+		// NB: Although index 15 is not inserted to the cache,
+		// we still return term = 7. Callers should make sure
+		// that index <= lastIndex when calling term(index).
+		{15, 7, true}, // beyond cache
+	}
+
+	for _, tcase := range testCases {
+		term, ok := tc.term(tcase.index)
+		require.Equal(t, tcase.wantTerm, term, "index %d", tcase.index)
+		require.Equal(t, tcase.wantOK, ok, "index %d", tcase.index)
+	}
+}
+
+func TestTermCachePopulatedTruncateAndAppend(t *testing.T) {
+	testCases := []struct {
+		name            string
+		cacheSize       uint64
+		initialID       entryID
+		initTerms       []uint64 // Initial terms to populate cache
+		appendInitialID entryID
+		appendTerms     []uint64 // Terms to append after initial population
+		want            struct {
+			cacheLen             int
```

Comment:
Probably no much sense asserting on this one. Focus is on visible behaviour of this struct.